### PR TITLE
GH-732: Fix User membership recognition bug

### PR DESCRIPTION
**Pull request by albertparedandan**

### CODE REVIEW

File: ui/src/components/admin/users/AdminUser.js [hunk 1]
```diff
@@ -200,21 +173,18 @@ export default function AdminUser(props) {
...
+  const membershipBaptismDateFormatter = (params) => {
+    if (params.value) {
+      const dateTimeFormat = 'dd MMM yyyy';
+      const dateTimeObj = DateTime.fromISO(params.value);
+      if (dateTimeObj.isValid)
+        return dateTimeObj.toFormat(dateTimeFormat);
     }
 
-    return '';
+    return ''
```
Comment (tamarayustian) [inline ui/src/components/admin/users/AdminUser.js:184]: minor, just missing semi-colon. no big deal

File: ui/src/components/admin/users/AdminUser.js [hunk 2]
```diff
@@ -444,9 +414,12 @@ export default function AdminUser(props) {
       ...data,
     });
 
-    if (res.status !== 200) {
+    if (res.status !== 200 || res.data.id == null || res.data.id === '') {
```
Comment (tamarayustian) [inline ui/src/components/admin/users/AdminUser.js:417]: why for the comparison of res.data.id, you used 2 different equal signs: '==' and '==='?  just curious. i dont think it affects anything

File: server/api/controllers/memberships/update-membership.js [hunk 3]
```diff
@@ -26,10 +26,26 @@ module.exports = {
...
-    const { id: membershipId, ...toUpdate } = params;
+    
+    const { id: membershipId, recommitmentDate, ...toUpdate } = params;
     if (membershipId) {
       try {
+        if (recommitmentDate && recommitmentDate !== '') {
+          let data = await Membership.findOne({ _id: membershipId });
+
+          if (data == null)
+            return exits.invalid(`No MembershipId found id: ${membershipId}`);
```
Comment (tamarayustian) [inline server/api/controllers/memberships/update-membership.js:37]: error msg can just be 'No MembershipId found' to be grammatically correct

**Review by tamarayustian — APPROVED** [review]

File: ui/src/components/admin/users/AdminUser.js (hunk 2 above)
Comment (albertparedandan) [reply ui/src/components/admin/users/AdminUser.js:417]: `== null` checks if `id` is `null` or `undefined`, `=== '"` checks if `id` is string type and is not empty string

**Pull request merged**